src/axom/inlet/Field.cpp

```diff
@@ -64,6 +64,126 @@ bool Field::required()
   return (bool)intValue;
 }
 
+std::shared_ptr<Field> Field::addDefaultString(std::string value) {
```

Comment:
I think it would be cleaner to change these to 4 overloaded functions like defaultValue(std::string value) then check if m_type matches just like in your assert but throw a warning if its wrong.